### PR TITLE
Brms 213

### PR DESCRIPTION
**Pull request by chris-ianson**
Updated controller tests (updated test data to include missing keys)
Refactored Features class
Added groni feature toggle (set to false)

### CODE REVIEW

File: app/uk/gov/hmrc/brm/config/Features.scala [hunk 1]
```diff
@@ -102,59 +106,74 @@ trait FeatureFactory {
...
+
+  override def isDetailsMatchingEnabled(implicit p: Payload): Boolean = {
+    detailsFeature
   }
 
-  override def isReferenceMatchingEnabled(implicit p: Payload) : Boolean = {
-    true
+  override def isReferenceMatchingEnabled(implicit p: Payload): Boolean = {
+    referenceFeature
   }
```
Comment (adamconder) [inline app/uk/gov/hmrc/brm/config/Features.scala]: @chris-ianson Do we need to have this additional method `def referenceFeature: Boolean = GRONIReferenceFeature().enabled` we can just call the instance `GRONIReferenceFeature().enabled()` within `override def isDetailsMatchingEnabled(implicit p: Payload):` ?

File: test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala [hunk 2]
```diff
@@ -497,18 +484,6 @@ class BirthEventsControllerSpec
           header(ACCEPT, result).get shouldBe "application/vnd.hmrc.1.0+json"
         }
 
-        "return InternalServerError when GRO throws Exception" in {
-          when(mockAuditConnector.sendEvent(Matchers.any())(Matchers.any(), Matchers.any())).thenReturn(Future.successful(AuditResult.Success))
-          when(MockController.service.groConnector.getReference(Matchers.any())(Matchers.any())).thenReturn(Future.failed(new Exception("")))
-
```
Comment (adamconder) [inline test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala]: @chris-ianson Do we test this now?

**Review by adamconder — CHANGES_REQUESTED** [review]

File: test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala (hunk 2 above)
Comment (chris-ianson) [reply test/uk/gov/hmrc/brm/controllers/BirthEventsControllerSpec.scala]: My mistake, I removed this thinking it was a duplicate but it's getDetails what's getting tested. I've reinserted the test for getReference.

File: app/uk/gov/hmrc/brm/config/Features.scala (hunk 1 above)
Comment (chris-ianson) [reply app/uk/gov/hmrc/brm/config/Features.scala]: I've removed these from the interface and the other concrete implementations. Called directly now.

**Comment by chris-ianson** [comment]
Updated

**Review by chris-ianson — COMMENTED** [review]
Updated

**Review by adamconder — APPROVED** [review]

**Pull request merged**